From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Tue, 13 Oct 2026 00:00:00 -0500
Subject: [PATCH] Bug 1971509 - Disable panel opacity transition on non-mac
 when reduced motion is enabled

On Windows/GTK, prefers-reduced-motion is derived from a more general
"enable animations" setting (Animation Effects / gtk-enable-animations,
respectively), so disable the panel opacity transition there too even though
it is not quite a motion effect.

Origin: upstream, https://github.com/mozilla-firefox/firefox/commit/85b95fd9647e
Forwarded: not-needed
Last-Update: 2026-10-13
---
 toolkit/themes/shared/popup.css | 24 +++++++++++++-----------
 1 file changed, 13 insertions(+), 11 deletions(-)

diff --git a/toolkit/themes/shared/popup.css b/toolkit/themes/shared/popup.css
index 1163205446ca2..06812c762430c 100644
--- a/toolkit/themes/shared/popup.css
+++ b/toolkit/themes/shared/popup.css
@@ -125,21 +125,23 @@ panel[type="arrow"] {
     transition-duration: 0.18s;
   }
 
-  @media not (-moz-platform: macos) {
+  /* On Windows / GTK, prefers-reduced-motion is derived from a more general
+   * "enable animations" setting (Animation Effects / gtk-enable-animations
+   * respectively). So we disable the opacity transition there too even though
+   * it's not quite a motion effect. */
+  @media (not (-moz-platform: macos)) and (not (prefers-reduced-motion)) {
     transition-property: transform, opacity;
     will-change: transform, opacity;
     opacity: 0;
-    @media not (prefers-reduced-motion) {
-      transform: translateY(-70px);
+    transform: translateY(-70px);
 
-      &:where([side="bottom"]) {
-        transform: translateY(70px);
-      }
-      /* On some platforms we can't animate the panel position easily, see
-       * bug 1385079, bug 1800368, bug 1800442 */
-      @media not (-moz-panel-animations) {
-        transform: none;
-      }
+    &:where([side="bottom"]) {
+      transform: translateY(70px);
+    }
+    /* On some platforms we can't animate the panel position easily, see
+    * bug 1385079, bug 1800368, bug 1800442 */
+    @media not (-moz-panel-animations) {
+      transform: none;
     }
   }
 
